src/main/java/me/qoomon/maven/gitversioning/GitVersioningModelProcessor.java

```diff
@@ -290,13 +335,52 @@ private Model processModel(Model projectModel) throws IOException {
                         sessionProjectDirectories.add(new File(projectModel.getProjectDirectory(), module).getCanonicalPath());
                     }
                 }
+
             }
 
             this.virtualProjectModelCache.put(projectId, virtualProjectModel);
+
+            for (Model model : virtualProjectModelCache.values()) {
```

Comment:
there is no need to loop over the whole project cache for each project.
the processor is called for each project, so you need to do this only for the current project `virtualProjectModel `.